ui/app/src/components/RenameContentDialog/RenameContentDialogContainer.tsx

```diff
@@ -0,0 +1,123 @@
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { RenameContentDialogProps } from './RenameContentDialog';
+import React, { useState } from 'react';
+import { ensureSingleSlash, isBlank } from '../../utils/string';
+import { RenameItemView } from '../RenameDialogBody';
+import { DialogFooter } from '../DialogFooter';
+import SecondaryButton from '../SecondaryButton';
+import { FormattedMessage } from 'react-intl';
+import PrimaryButton from '../PrimaryButton';
+import { useEnhancedDialogContext } from '../EnhancedDialog';
+import { DetailedItem } from '../../models';
+import { DialogBody } from '../DialogBody';
+import { AjaxError } from 'rxjs/ajax';
+import useDebouncedInput from '../../hooks/useDebouncedInput';
+import { checkPathExistence } from '../../services/content';
+import useActiveSiteId from '../../hooks/useActiveSiteId';
+import useUpdateRefs from '../../hooks/useUpdateRefs';
+import { applyContentNameRules } from '../../utils/content';
+
+export interface RenameContentDialogContainerProps
+  extends Pick<
+    RenameContentDialogProps,
+    'path' | 'value' | 'onRenamed' | 'onClose' | 'onSubmittingAndOrPendingChange'
+  > {
+  dependantItems: DetailedItem[];
+  fetchingDependantItems: boolean;
+  error: AjaxError;
+}
+
+export function RenameContentDialogContainer(props: RenameContentDialogContainerProps) {
+  const {
+    path,
+    value,
+    onRenamed,
+    onClose,
+    dependantItems,
+    fetchingDependantItems,
+    error,
+    onSubmittingAndOrPendingChange
+  } = props;
+  const isPage = value.includes('/index.xml');
+  const { isSubmitting } = useEnhancedDialogContext();
+  const strippedValue = isPage ? value.replace('/index.xml', '') : value.replace('.xml', '');
+  const [name, setName] = useState(strippedValue);
+  const [itemExists, setItemExists] = useState(false);
+  const isValid = !isBlank(name) && !itemExists && name !== strippedValue;
+  const [confirmBrokenReferences, setConfirmBrokenReferences] = useState(false);
+  const renameDisabled =
+    isSubmitting || !isValid || fetchingDependantItems || (dependantItems?.length > 0 && !confirmBrokenReferences);
+  const siteId = useActiveSiteId();
+  const refs = useUpdateRefs({ onNameUpdate$: null });
+
+  refs.current.onNameUpdate$ = useDebouncedInput((name: string) => {
+    checkPathExistence(siteId, `${ensureSingleSlash(`${path}/${name}`)}${isPage ? '/index.xml' : '.xml'}`).subscribe(
+      (exists) => {
+        setItemExists(name !== strippedValue && exists);
+      }
+    );
+  }, 400);
```

Comment:
Don't really see the need for this as a ref.

```suggestion
  const refs = useUpdateRefs({

onNameUpdate$: useDebouncedInput((name: string) => {
    checkPathExistence(siteId, `${ensureSingleSlash(`${path}/${name}`)}${isPage ? '/index.xml' : '.xml'}`).subscribe(
      (exists) => {
        setItemExists(name !== strippedValue && exists);
      }
    );
  }, 400)

});
```

Reply:
Yeah not needed, was initially supposed to be used differently. Removing ref